forc-plugins/forc-node/src/run_opts.rs

```diff
@@ -0,0 +1,155 @@
+use std::{fmt::Display, path::PathBuf};
+
+/// Possible parameters to set while integrating with `fuel-core run`.
+#[derive(Debug, Default)]
+pub struct RunOpts {
+    pub(crate) service_name: Option<String>,
+    /// DB type, possible options are: `["in-memory", "rocksdb"]`.
+    pub(crate) db_type: DbType,
+    /// Should be used for local development only. Enabling debug mode:
+    /// - Allows GraphQL Endpoints to arbitrarily advance blocks.
+    /// - Enables debugger GraphQL Endpoints.
+    /// - Allows setting `utxo_validation` to `false`.
+    pub(crate) debug: bool,
+    /// Snapshot from which to do (re)genesis.
+    pub(crate) snapshot: PathBuf,
+    /// Peering private key from generated key-pair.
+    pub(crate) keypair: Option<String>,
+    /// Ethereum RPC endpoint.
+    pub(crate) relayer: Option<String>,
+    /// The IP address to bind the GraphQL service to.
+    pub(crate) ip: Option<std::net::IpAddr>,
+    /// The port to bind the GraphQL service to.
+    pub(crate) port: Option<u16>,
+    /// p2p network's TCP port.
+    pub(crate) peering_port: Option<u16>,
+    /// The path to the database, only relevant if the db type is not
+    /// "in-memory".
+    pub(crate) db_path: Option<PathBuf>,
```

Comment:
Maybe struct should be embedded in the `DbType` enum? i.e. `DbType::RocksDB(Option<Pathbuf>)`
^ unless this is not supported by clap.

Reply:
Not sure if this is supported